participants/l10n_update_ts.py

```diff
@@ -222,42 +222,58 @@ def update_ts(self, tsrpms, packagename, version, tmpdir, branch):
 
     def init_gitdir(self, gitdir, reponame, branch):
         """Initialize local clone of remote Git repository."""
+        remote_branch = "origin/%s" % branch
 
-        if not os.path.exists(gitdir):
-            # check if repo exists on git server
-            gitserv_auth = (self.gitconf["username"], self.gitconf["password"])
-            ghresp = requests.get(
-                "%s/user/repos" % self.gitconf["apiurl"], auth=gitserv_auth
-            )
-            if reponame not in [repo['name'] for repo in ghresp.json()]:
-                payload = {
-                    'name': reponame,
-                    'has_issues': False,
-                    'has_wiki': False,
-                    'has_downloads': False,
-                    'auto_init': True
-                }
-                ghresp = requests.post(
-                    "%s/user/repos" % self.gitconf["apiurl"],
-                    auth=gitserv_auth,
-                    headers={'content-type': 'application/json'},
-                    data=json.dumps(payload),
-                )
-                ghresp.raise_for_status()
+        if os.path.exists(gitdir):
+            check_call(["git", "fetch"], cwd=gitdir)
+            check_call(["git", "reset", "--hard", remote_branch], cwd=gitdir)
+            return
```

Comment:
So for the case where the repo is first cloned the branches are checked and a new branch may be created using git checkout --force -b ... so why isn't that done here too? :)

Reply:
because there is a gitdir *per branch* ... ugh :D 
OK